atomic_reactor/plugins/build_orchestrate_build.py

```diff
@@ -185,22 +185,29 @@ def __init__(self, tasker, workflow, platforms, build_kwargs,
 
         self.worker_builds = []
 
-    def get_excluded_platforms(self):
-        df_dir = self.workflow.source.get_dockerfile_path()[1]
-        exclude_platforms = set()
-        exclude_path = os.path.join(df_dir, self.EXCLUDE_FILENAME)
-        if os.path.exists(exclude_path):
-            with open(exclude_path) as f:
-                for platform in f:
-                    platform = platform.strip()
-                    if not platform:
-                        continue
-                    exclude_platforms.add(platform)
-
-        return exclude_platforms
-
     def get_platforms(self):
-        return self.platforms - self.get_excluded_platforms()
+        df_dir = self.workflow.source.get_dockerfile_path()[1]
+        excluded_platforms = set()
+        container_path = os.path.join(df_dir, self.CONTAINER_FILENAME)
+        if os.path.exists(container_path):
+            with open(container_path) as f:
+                data = yaml.load(f)
+                if data is None or 'platforms' not in data or data['platforms'] is None:
+                    return self.platforms
+                for action, platforms in data['platforms'].items():
+                    if action == 'not':
```

Comment:
There's some duplication here when processing the platforms value. Consider adding a helper method to ensure that value is a list:
```python
def make_list(self, value):
    if not isinstance(value, list):
        value = [value]
    return value
```

I also think that looping over the actions is not really necessary since we know which keys we're looking for:

```python
excluded_platforms = set(make_list(data['platforms'].get('not', [])))
self.platforms += set(make_list(data['platforms'].get('only', [])))
```